test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -215,9 +215,9 @@ void UcxContext::progress(unsigned count)
 {
     int i = 0;
 
-    while ((i++ < count) && (ucp_worker_progress(_worker) > 0)) {
+    do {
         progress_io_message();
-    }
+    } while (ucp_worker_progress(_worker) && (++i < count));
```

Comment:
BTW maybe like this:
```
ucp_worker_progress(_worker);
do {
         progress_io_message();
} while ((++i < count) && ucp_worker_progress(_worker));
```

to make sure we progress once before checking io_message
otherwise - it will be checked only in next loop

Reply:
IMO
if `count > 1` - it does not matter, we should be in tight `while` loop
if `count == 1` - the ordering of `progress_io_message()` <-> `ucp_worker_progress()` should not matter as well

Reply:
> if `count == 1` - the ordering of `progress_io_message()` <-> `ucp_worker_progress()` should not matter as well

you mean it doesn't matter if we just polled new io_message and will actually process it only in the next loop?

Reply:
yes

Reply:
IMO it would be slightly better for latency to process it in same loop

Reply:
ok, up to you